master/static/srv/get_active_allocations.sql

```diff
@@ -0,0 +1,13 @@
+WITH activeAllocations AS (
+  SELECT state, task_id
+  FROM allocations
+  WHERE state IN ('PULLING', 'RUNNING', 'STARTING')
+)
+SELECT trials.task_id AS TaskID,
+  trials.id AS TrialID,
+  trials.experiment_id AS ExperimentID,
+  (SELECT COUNT(*) > 0 FROM activeAllocations WHERE activeAllocations.task_id = trials.task_id AND state = 'RUNNING') AS IsRunning,
+  (SELECT COUNT(*) > 0 FROM activeAllocations WHERE activeAllocations.task_id = trials.task_id) AS IsStarting
+FROM trials
+WHERE ($1 = 0 OR trials.id = $1)
+  AND ($2 = 0 OR trials.experiment_id = $2);
```

Comment:
this is going to be an expensive query with that CTE on some Postgres versions. can you rewrite as a join?

Reply:
also, maybe write it in terms of jobs/tasks instead of exp/trials

Reply:
✅  made an update using tasks table and JOIN